File: packages/python/src/tesote_sdk/v2/status.py
"""v2 status / whoami endpoints (`/v2/status`, `/v2/whoami`)."""

from __future__ import annotations

from typing import Any, Dict

from ..models import StatusResponse, WhoAmI
from ..transport import Transport


class StatusResource:
    """`/v2/status` (no auth) and `/v2/whoami` (auth)."""

    def __init__(self, transport: Transport) -> None:
        self._transport = transport

    def status(self) -> StatusResponse:
        response = self._transport.request("GET", "/v2/status")
        body: Dict[str, Any] = response.json if isinstance(response.json, dict) else {}
        return StatusResponse.from_dict(body)

    def whoami(self) -> WhoAmI:
        response = self._transport.request("GET", "/v2/whoami")
        body: Dict[str, Any] = response.json if isinstance(response.json, dict) else {}
        return WhoAmI.from_dict(body)


__all__ = ["StatusResource"]

File: packages/ruby/lib/tesote_sdk/v2/status.rb
module TesoteSdk
  module V2
    class Status
      def initialize(transport)
        @transport = transport
      end

      # GET /v2/status — note this lives at /api/v2/status (not /api/status).
      def status(opts: {})
        body = @transport.request('GET', 'status', opts: opts)
        Models::StatusResult.from_hash(body)
      end

      # GET /v2/whoami
      def whoami(opts: {})
        body = @transport.request('GET', 'whoami', opts: opts)
        Models::Whoami.from_hash(body)
      end
    end
  end
end
